webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,254 @@
+  );
+
+describe("PerturbationTestingPreview", () => {
+  const handlers = [getPertubationResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should be having two toggle buttons for the train and test", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      expect(screen.getAllByRole("button")).toHaveLength(2);
+      expect(screen.getAllByRole("button")).not.toHaveLength(1);
+    });
+  });
+
+  it("should be having a disabled toggle button if either one of the datasplit is unavailable", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => {
+      //Verifying if the train toggle button is disabled if not available in datasplits
+      expect(screen.getByRole("button", { pressed: false })).toHaveClass(
+        "Mui-disabled"
+      );
+      // By default, evaluation toggle is selected on page load.
+      expect(screen.getByRole("button", { pressed: true })).not.toHaveClass(
+        "Mui-disabled"
+      );
+    });
+  });
+
+  it("should display the data in the toggle with expected styling (color, font), format", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      // Verrify if the actual response data from API is visibly displayed on the toggle
+      expect(screen.getByText(/14.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Evaluation Set")).toBeVisible();
+      expect(screen.getByText(/16.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Training Set")).toBeVisible();
+
+      // styling should be applied for the selected toggle as expected
+      expect(screen.getByText(/14.0%/)).toHaveStyle(
+        "font-size: 3.75rem; color: rgb(237, 108, 2)"
+      );
+
+      // selected toggle should be highlighted with some bgcolor
+      expect(screen.getByRole("button", { pressed: true })).toHaveStyle(
+        "color: rgb(156, 39, 176)"
+      );
+
+      // verify with some random unexpected values to not have been displayed.
+      expect(screen.queryByText(/17.0%/i)).toBeNull();
+      expect(screen.queryByText("some random text")).toBeNull();
+      expect(screen.queryByText(/26.0%/i)).toBeNull();
+    });
+  });
+
+  it("should not display value if a datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => expect(screen.getByText("--%")).toBeVisible());
+  });
+
+  it("should have expected columns", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    const table_columns = [
+      "Test Family",
+      "Test Name",
+      "Modif. Type",
+      "Test Description",
+      "FR on Evaluation Set",
+    ];
+    // verify if all the required columns are displayed
+    waitFor(() =>
```

Comment:
You are not testing anything if you don't `await` the promise. I see about 10 other omissions in this file.
```suggestion
    await waitFor(() =>
```